packages/cadl-ranch-specs/http/azure/core/traits/mockapi.ts

```diff
@@ -33,3 +33,44 @@ Scenarios.Azure_Core_Traits_smokeTest = passOnSuccess(
     };
   }),
 );
+
+Scenarios.Azure_Core_Traits_repeatableAction = passOnSuccess(
+  mockapi.post("/azure/core/traits/user/:id:repeatableAction", (req) => {
+    if (req.params.id !== "1") {
+      throw new ValidationError("Expected path param id=1", "1", req.params.id);
+    }
+
+    if (!("repeatability-request-id" in req.headers)) {
+      throw new ValidationError("Repeatability-Request-ID is missing", "A UUID string", undefined);
+    }
+    if (!("repeatability-first-sent" in req.headers)) {
+      throw new ValidationError("Repeatability-First-Sent is missing", "A date-time in headers format", undefined);
+    }
+
+    if (!/^[0-9a-f]{8}-([0-9a-f]{4}-){3}[0-9a-f]{12}$/i.test(req.headers["repeatability-request-id"])) {
```

Comment:
can we move this to an helper and reuse it in the `special-headers/repeatability` too.

Reply:
@timotheeguerin I've moved the validation to `cadl-ranch-api`, could you help to review?